lib/internal/streams/transform.js

```diff
@@ -244,3 +246,23 @@ Transform.prototype._read = function() {
     callback();
   }
 };
+
+Transform.by = function by(asyncGeneratorFn, options) {
+  let _resolve;
+  let _promise = new Promise((resolve) => _resolve = resolve);
```

Comment:
You can use `createDeferredPromise` from `internal/util`
```suggestion
  let { promise, resolve } = createDeferredPromise();
```